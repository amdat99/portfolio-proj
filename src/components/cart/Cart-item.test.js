import React from 'react';

import { shallow,render,mount } from 'enzyme';
import {createSerializer} from 'enzyme-to-json';
import toJson from 'enzyme-to-json';
import { withRouter } from 'react-router-dom'

import CartItem from './Cart-item'

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));
// describe('cart item', () => {
      
        let mockFetchProductPending = jest.fn();
        let  mockIncrementItem = jest.fn();
        let mockDecrementItem = jest.fn();
        let wrapper
        let wrapper2

        
  beforeEach(() => {

   const mockProps = {
            item: {
              picture: 'www.picture.com',
              price: 10,
              name: 'ball',
              quantity: 5,
              profileId: 123
            },
            fetchProductPending: mockFetchProductPending,
            incrementItem: mockIncrementItem,
            decrementItem: mockDecrementItem
          }; 
          
           wrapper = toJson(shallow(<CartItem  {...mockProps} />))
           wrapper2 = shallow(<CartItem {...mockProps} />);
        })
        



it('expect to render cartItem component', () => {
    expect(wrapper).toMatchSnapshot();
})

it('expect increment item to be called item added', () => {

    wrapper2.find('img').simulate('click')
    expect(mockFetchProductPending).toHaveBeenCalled();
  });
// 