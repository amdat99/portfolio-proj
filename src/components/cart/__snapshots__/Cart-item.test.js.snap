// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render cartItem component 1`] = `
<div
  className="cart-item"
>
  <Link
    to="/product"
  >
    <img
      alt="item"
      className="imagetest"
      onClick={[Function]}
      src="www.picture.com?100x75"
      style={
        Object {
          "cursor": "pointer",
        }
      }
      width="75"
    />
  </Link>
  <div
    className="cartitem-details"
  >
    <span
      className="cartitem-name"
    >
      ball
    </span>
    <div
      className="cartitem-name"
    >
      5
      xÂ£
      10
    </div>
    <div
      className="cartitem-remove"
    >
      <span
        className="cartitem-remove-span "
        onClick={[Function]}
      >
        remove 
      </span>
      <span
        className="cartitem-remove-span "
        onClick={[Function]}
      >
        add
      </span>
    </div>
  </div>
</div>
`;
